### ISSUE TITLE
tsconfig paths not working

### ISSUE THREAD

**Issue by splincode** [piotrwitek/ts-mocha#25]
tsconfig.json

```
{
    "compilerOptions": {
        
        "paths": {
            "@logger/config": [
                "logger.config.ts"
            ]
        }
    },
 
}
```

index.ts

```
import * from "@logger/config"
...
```

> ts-mocha -p tsconfig.json src/**/*.spec.ts

Error: Cannot find module '@logger/config'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
    at Function.Module._load (internal/modules/cjs/loader.js:520:25)
    at Module.require (internal/modules/cjs/loader.js:650:17)

**Comment by piotrwitek** [comment]
Hey @splincode 
Are you aware how paths mapping works in typescript?
How do you imaging this issue could be solved on this library end, as this is only a wrapper on mocha. Do you think it is feasible to put responsibility of parsing and rewriting module imports to this library?

**Comment by splincode** [comment]
@piotrwitek it would be great if you added tsconfig-paths from out of the box
https://www.npmjs.com/package/tsconfig-paths

**Comment by piotrwitek** [comment]
@splincode nice this library seems like a really good fit for this use-case

We will need to create a POC and add a few test cases, then I would be able to merge it.

Don't know if I'll be able to find time to work on it anytime soon, but I'll accept PR if someone is interested to tackle it.

**Comment by strantr** [comment]
@splincode Hey, I took a quick look at this and it seems to be working fine. Are there any test cases you were thinking of in particular?

**Comment by piotrwitek** [comment]
Hey @strantr, that's cool!

The easiest way I was thinking about would be something like this:
1) I would start by adding `tsconfig-paths` package as a dev dependency and peer dependency only (set the correct version range)
2) Then restructuring test folder to have sub-folders:
  - `baseline` sub-folder for old tests
  - `paths` sub-folder for new tests (please use an old folder contents as a starting point)
3) edit tsconfig.json in paths subfolder to include paths setup from above example or your project, whatever works and is simple to understand by other contributors
4) update relative paths setup in paths version of this file: https://github.com/piotrwitek/ts-mocha/blob/master/test/programmatic-use-test.js
4) update test command to run tests from both folders (baseline and paths)
  - https://github.com/piotrwitek/ts-mocha/blob/master/package.json#L18

Good luck!

**Comment by charlesbodman** [comment]
Works!

**Comment by piotrwitek** [comment]
To finalize this feature still need to add a new docs section: https://github.com/piotrwitek/ts-mocha/pull/26#issuecomment-408634193

I will take care of this in a few minutes.

**Comment by piotrwitek** [comment]
I changed `tsconfig-paths` to be opt-in as a non-breaking feature.

Closed with #27 and release of v1.3.0 on npm.

Feedback appreciated!
